File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();

        for(String str: strs){
            char[] freqMap = new char[26];
            for(char c: str.toCharArray()){
                freqMap[c - 'a']++; 
            }
            String key = String.valueOf(freqMap);

            List<String> temp = new ArrayList<>();
            if(map.containsKey(key)){
                temp = map.get(key);
            }
            temp.add(str);
            map.put(key, temp);

        }
        return new ArrayList(map.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        dic = {}
        for s in strs:
            sortedVal = tuple(sorted(s))
            if sortedVal not in dic:
                dic[sortedVal] = []

            dic[sortedVal].append(s)

        return list(dic.values())

